### PR TITLE
RavenDB-1127: RavenDB-11137: Allow failure before 'Send' success afte…

### PR DESCRIPTION
**Pull request by aviviadi**
…r DB unload

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/Operation.cs [hunk 1]
```diff
@@ -54,12 +54,25 @@ public Operation(RequestExecutor requestExecutor, Func<IDatabaseChanges> changes
             var observable = changes.ForOperationId(_id);
             _subscription = observable.Subscribe(this);
             await observable.EnsureSubscribedNow().ConfigureAwait(false);
-
+            changes.ConnectionStatusChanged += ConnectionStatusEvent;
             await FetchOperationStatus().ConfigureAwait(false);
         }
 
+        private async void ConnectionStatusEvent(object sender, EventArgs e)
```
Comment (ayende) [inline src/Raven.Client/Documents/Operations/Operation.cs:61]: Let's avoid `async void` and friends.
And we need to call to `TrySetException` in this case, no?

Comment (ppekrol) [reply src/Raven.Client/Documents/Operations/Operation.cs:61]: @aviviadi what do you think about this approach: https://github.com/ppekrol/ravendb/commits/RavenDB-11137 ?

Comment (aviviadi) [reply src/Raven.Client/Documents/Operations/Operation.cs:61]: @ayende - if we TrySetException, we might report failure on a successful operation. Hunged client (with hopefully timeout set) is more reasonable for operation with no knowledge if it went all the way or not (and the connection is probably "dead" anyhow at this point)

Comment (aviviadi) [reply src/Raven.Client/Documents/Operations/Operation.cs:61]: @ppekrol - obviously that's better (fixes the above remark)

Comment (ayende) [reply src/Raven.Client/Documents/Operations/Operation.cs:61]: Yes, let's go ahead and merge this

**Pull request merged**